model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -66,25 +68,55 @@ data class PackageConfiguration(
      * License finding curations.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList()
+    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList(),
+
+    /**
+     * The source code origin this configuration applies to.
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val sourceCodeOrigin: SourceCodeOrigin? = null
 ) {
     init {
-        require((sourceArtifactUrl == null) xor (vcs == null)) {
-            "A package configuration must either set the 'sourceArtifactUrl' or the 'vcs' property."
+        val vcsOrSourceArtifact = (sourceArtifactUrl == null) xor (vcs == null)
+        require(
+            vcsOrSourceArtifact || (sourceCodeOrigin != null && vcs == null) ||
+                (sourceCodeOrigin == null && vcs == null && sourceArtifactUrl == null)
```

Comment:
This should become: `(sourceArtifactUrl == null) xor (vcs == null) xor (sourceCodeOrigin == null)`

Reply:
No, this does not work: `xor` with three inputs does not behave like one would assume.

If `sourceArtifactUrl != null`  and `vcs != null`, `(sourceArtifactUrl == null) xor (vcs == null)` is `false`.

Therefore if `sourceCodeOrigin == null`, `(sourceArtifactUrl == null) xor (vcs == null) xor (sourceCodeOrigin == null)` would return `true`, which is wrong: you would accept a package configuration with both `sourceArtifactUrl` and `vcs` set.